zenoh/src/net/routing/interceptor/authorization.rs

```diff
@@ -195,19 +197,51 @@ impl PolicyEnforcer {
     */
     pub fn policy_information_point(
         &self,
-        config_rule_set: &Vec<AclConfigRules>,
+        config_rule_set: &[AclConfigRules],
     ) -> ZResult<PolicyInformation> {
         let mut policy_rules: Vec<PolicyRule> = Vec::new();
-        for config_rule in config_rule_set {
-            for subject in &config_rule.interfaces {
-                for flow in &config_rule.flows {
-                    for action in &config_rule.actions {
-                        for key_expr in &config_rule.key_exprs {
+        for (rule_offset, config_rule) in config_rule_set.iter().enumerate() {
+            // config validation
+            let mut validated_rule = config_rule.clone();
+            let mut validation_err = String::new();
+            if validated_rule.interfaces.is_empty() {
+                tracing::warn!("ACL config interfaces list is empty. Applying rule #{} to all network interfaces", rule_offset);
```

Comment:
I believe this is not the right place to implement such behaviour. `policy_information_point` should work on valid configuration and it's completely fine to consider an empty interface list as invalid here. Proper enforcement should be done at configuration file parsing.